react-components/src/hooks/use3dScenes.tsx

```diff
@@ -58,25 +61,36 @@ export const use3dScenes = (
           const spaceMap = acc[space];
 
           if (spaceMap[externalId] === undefined) {
-            spaceMap[externalId] = [];
+            spaceMap[externalId] = { name: '', modelOptions: [] };
           }
           const externalIdMap = spaceMap[externalId];
-          externalIdMap.push(newModel);
+          externalIdMap.modelOptions.push(newModel);
 
           return acc;
         },
         // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-        {} as Record<Space, Record<ExternalId, AddReveal3DModelOptions[]>>
+        {} as Record<Space, Record<ExternalId, SceneData>>
       );
 
+      scenesQueryResult.items.scenes.forEach((scene) => {
+        const { space, externalId } = scene;
+
+        if (scenesMap[space]?.[externalId] !== undefined) {
+          const properties = Object.values(
+            Object.values(scene.properties)[0] as Record<string, unknown>
+          )[0] as SceneConfigurationProperties;
+          scenesMap[space][externalId].name = properties.name;
+        }
+      });
```

Comment:
I would like this to be put in its own function and be done before the `reduce` above.. In other words - this block creates the map from space-externalId to scene object with name, and the reduce-block fills it with models

Reply:
Fixed in [d11b2b7](https://github.com/cognitedata/reveal/pull/4131/commits/d11b2b779ae03f129537389b15adf9d446582de0) along with some refactoring